### PR TITLE
feat(mantis): demo order contract and fixes for no-std support

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Required for merge:
- [ ] `pr-workflow-check / draft-release-check` is ✅ success
- Other rules GitHub shows you, or can be read in [configuration](../terraform/github.com/branches.tf) 

Makes review faster:
- [ ] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [ ] I marked PR by `misc` label if it should not be in release notes
- [ ] Linked Zenhub/Github/Slack/etc reference if one exists
- [ ] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [ ] Added reviewer into `Reviewers`
- [ ] I tagged(`@`) or used other form of notification of one person who I think can handle best review of this PR
- [ ] I have proved that PR has no general regressions of relevant features and processes required to release into production
- [ ] Any dependency updates made, was done according guides from relevant dependency
- Clicking all checkboxes 
- Adding detailed description of changes when it feels appropriate (for example when PR is big)

### CODE REVIEW

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
 
-use crate::error::ContractError;
-
-pub type Amount = Displayed<u128>;
-pub type OrderId = Displayed<u128>;
+// fix core in std in contract
+// use xc_core::{service::dex::ExchangeId, shared::Displayed, NetworkId};
+pub type ExchangeId = Uint128;
+pub type Amount = u128;
+pub type OrderId = Uint128;
```
Comment (mina86) [inline code/xcvm/cosmwasm/contracts/order/src/lib.rs:22]: I find this confusing. Why is ExchangeId set to Uint128 while Amount is u128? It seems to me that those aliases should all either be plain Rust type or Uint128 and then in structs we should accordingly either use Uint128 or those aliases.

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs [hunk 2]
```diff
@@ -124,15 +124,18 @@ pub struct SolvedOrder {
 impl SolvedOrder {
 	pub fn new(order: OrderItem, solution: Cow) -> StdResult<Self> {
 		ensure!(
-			order.msg.wants.amount.u128() >= solution.given.0,
-			StdError::GenericErr { msg: "user limit was not satisfied".to_string() }
+			order.msg.wants.amount.u128() <= solution.given.u128(),
```
Comment (mina86) [inline code/xcvm/cosmwasm/contracts/order/src/lib.rs:127]: Wouldn’t the following work?

```suggestion
			order.msg.wants.amount <= solution.given,
```

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs [hunk 3]
```diff
@@ -124,15 +124,18 @@ pub struct SolvedOrder {
...
+			StdError::generic_err(format!(
+				"user limit was not satisfied {:?} {:?} ",
+				&order, &solution
```
Comment (mina86) [inline code/xcvm/cosmwasm/contracts/order/src/lib.rs:130]:
```suggestion
				"user limit was not satisfied {order:?} {solution:?} ",
```

**Review by mina86 — APPROVED** [review]

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/xcvm/cosmwasm/contracts/order/src/lib.rs:22]: Sylvia does not compile messages with u128, only Uint128.

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs (hunk 2 above)
Comment (dzmitry-lahoda) [reply code/xcvm/cosmwasm/contracts/order/src/lib.rs:127]: some time ago it was not working for sure, as they did no impl traits, did not checked now. would it?

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs (hunk 3 above)
Comment (dzmitry-lahoda) [reply code/xcvm/cosmwasm/contracts/order/src/lib.rs:130]: it uses references or clones? would it move objects into log. I am just not sure it compiles

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/xcvm/cosmwasm/contracts/order/src/lib.rs:22]: will try with displayed after making it compile in wasm std

Comment (mina86) [reply code/xcvm/cosmwasm/contracts/order/src/lib.rs:22]: Sure, the messages should use Uint128 but my point is that either consistently use Uint128 in the messages or declare those aliases to be Uint128 and use them. The way here some are Uint128 and some are u128 is confusing and misleading.

File: code/xcvm/cosmwasm/contracts/order/src/lib.rs (hunk 3 above)
Comment (mina86) [reply code/xcvm/cosmwasm/contracts/order/src/lib.rs:130]: Clones are never done implicitly and `format` always works on references. If you use `&order` you’re effectively using a double reference. It compiles to the same thing of course but just using it in format message is more concise.

**Review by mina86 — DISMISSED** [review]

**Pull request merged**